test/unit/CrossChainBaseTest.t.sol

```diff
@@ -79,14 +68,21 @@ contract CrossChainBaseTest is Test {
 
         {
             vm.selectFork(mainnetFork);
-            mainnetDeployer = new ImmutableMultiChainDeployer{salt: "SALT"}();
             mainnetERC20 = new ERC20Mock("Test Token", "TEST");
-            mainnetOFTAdapterImpl =
-                address(new L1YnOFTAdapterUpgradeable(address(mainnetERC20), address(mainnetLzEndpoint)));
+
+            bytes32 mainnetOFTAdapterSalt = createSalt(_deployer, "OFTAdapter");
+            bytes32 mainnetOFTAdapterProxySalt = createSalt(_deployer, "OFTAdapterProxy");
+
             mainnetOFTAdapter = L1YnOFTAdapterUpgradeable(
-                address(new TransparentUpgradeableProxy(mainnetOFTAdapterImpl, _controller, ""))
```

Comment:
deploy all contracts using `CREATE3Factory` instead of ImmutableMultichainDeployer